docs/articles/blog/2019-2-7-migration-from-testcafe-v0-x-y-to-v1-0-0.md

```diff
@@ -0,0 +1,147 @@
+        // ...
+    });
+}
+```
+
+**test.js**
+
+```js
+import runFixture from './external-lib';
+
+const fixtureName = 'My fixture';
+const url = 'https://testPage';
+
+runFixture(fixtureName, url);
+```
+
+## Programming Interface: Multiple Method Calls Prohibited
+
+Previous versions allowed you to call the [runner.src](https://devexpress.github.io/testcafe/documentation/using-testcafe/programming-interface/runner.html#src), [runner.browsers](https://devexpress.github.io/testcafe/documentation/using-testcafe/programming-interface/runner.html#browsers) and [runner.reporter](https://devexpress.github.io/testcafe/documentation/using-testcafe/programming-interface/runner.html#reporter) methods several times to specify multiple test files, browsers or reporters.
+
+```js
+const stream = fs.createWriteStream('report.json');
+
+runner
+    .src('/home/user/tests/fixture1.js')
+    .src('fixture5.js')
+    .browsers('chrome')
+    .browsers('firefox:headless')
+    .reporter('minimal')
+    .reporter('json', stream);
+```
+
+Starting with v1.0.0, pass arrays to these methods to specify multiple values.
+
+To use a reporter that writes to a file, add a `{ name, output }` object to an array (see the [runner.reporter](https://devexpress.github.io/testcafe/documentation/using-testcafe/programming-interface/runner.html#reporter) description for details).
+
+```js
+runner
+    .src(['/home/user/tests/fixture1.js', 'fixture5.js'])
+    .browsers(['chrome', 'firefox:headless'])
+    .reporter(['minimal', { name: 'json', output: 'report.json' }]);
+```
+
+### What Has Improved
+
+This change was necessary to implement the [configuration file](https://devexpress.github.io/testcafe/documentation/using-testcafe/configuration-file.html) in a way that is consistent with the API and command line interface.
+
+## Custom Request Hooks: Asynchronous API
+
+[Request hook](https://devexpress.github.io/testcafe/documentation/test-api/intercepting-http-requests/) methods became asynchronous in TestCafe v1.0.0.
+
+If the [onRequest](https://devexpress.github.io/testcafe/documentation/test-api/intercepting-http-requests/creating-a-custom-http-request-hook.html#the-onrequest-method) or [onResponse](https://devexpress.github.io/testcafe/documentation/test-api/intercepting-http-requests/creating-a-custom-http-request-hook.html#the-onresponse-method) method in your custom hook returns a Promise, TestCafe now waits for this Promise to resolve.
+
+This does not necessarily leads to unexpected behavior, but still be aware of possible side effects.
+
+Since the [onRequest](https://devexpress.github.io/testcafe/documentation/test-api/intercepting-http-requests/creating-a-custom-http-request-hook.html#the-onrequest-method) and [onResponse](https://devexpress.github.io/testcafe/documentation/test-api/intercepting-http-requests/creating-a-custom-http-request-hook.html#the-onresponse-method) methods are now asynchronous, add the `async` keyword to their declarations.
```

Comment:
Do not use 'Since'. There is no conclusion here.

Reply:
You should add the `async` keyword to the asynchronous [onRequest](https://devexpress.github.io/testcafe/documentation/test-api/intercepting-http-requests/creating-a-custom-http-request-hook.html#the-onrequest-method) and [onResponse](https://devexpress.github.io/testcafe/documentation/test-api/intercepting-http-requests/creating-a-custom-http-request-hook.html#the-onresponse-method) method declarations.